src/base/QXmppElement.cpp

```diff
@@ -123,10 +123,13 @@ QXmppElement::~QXmppElement()
 
 QXmppElement &QXmppElement::operator=(const QXmppElement &other)
 {
-    other.d->counter.ref();
-    if (!d->counter.deref())
-        delete d;
-    d = other.d;
+    if (this != &other) // self-assignment check
```

Comment:
1) Qt uses smart pointers (QSharedDataPointer, etc) in many classes.
2) There are few places in Qt there self assignment checks are present. For example:
http://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/tools/qcollator.cpp?h=v5.8.0#n107
http://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/io/qdebug.h?h=v5.8.0#n200
http://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/kernel/qvariant.cpp?h=v5.8.0#n1881
http://code.qt.io/cgit/qt/qtbase.git/tree/src/gui/text/qtextoption.cpp?h=v5.8.0#n122
etc.